### PR TITLE
restyle preferences payments

### LINKED ISSUE #5494: Update / restyle for Preferences > Payments
Starting off with a picture (courtesy of @bradleyrichter)
image

Style updates include:
- sortableTable no longer hides it's style (should properly show the column headers)
- re-connected styles for all widgets with master styles

New functionality:
- "hide excluded sites" toggle
- exclude button on the left (in addition to the existing right click one)
(for blacklisting sites that you don't want to see regardless of their include/exclude state)

### PR DESCRIPTION
**Pull request by jkup**
- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- [x] Ran `git rebase -i` to squash commits (if needed).

Fix #5494

Auditors @bsclifton @bradleyrichter 

Test Plan:

1. Open Brave
2. Go to Preferences > Payments
3. Toggle a site to not be included
4. Toggle hide excluded sites and make sure that one disappears
5. Make sure the general design looks like the mock in #5494
6. Make sure clicking the remove button for a site removes it from the list

Here is a screenshot of this branch:
screen shot 2016-12-08 at 2 20 43 pm

Spec image:
screenshot 2016-12-17 17 31 04

### CODE REVIEW

**Comment by jkup** [comment]
@bsclifton there is a seemingly related test failing on this build but it appears to be failing on master too

8) synopsis can disable site:
     Error: Promise was rejected with the following reason: TypeError: Cannot read property 'ledgerPayments' of undefined

Is this a known failure?

**Comment by bsclifton** [comment]
Just checked it out...

- The verified icon doesn't show properly
screen shot 2016-12-09 at 12 13 56 am

- Clicking remove does kill the entry. I can then go to shields and undo the removal... but the data appears to be lost for that. Is this intentional?  I know this PR is doing the same thing as when you right click and pick "Do not show this entry".  Maybe @mrose17 or @bradleyrichter can confirm this is correct

- The table looks great! :smile: love seeing it match the others

- The removal of the 60% style on sortableTable breaks the history page. It may have been sloppy of me to introduce that (instead of styling the columns individually).  All the other table locations look OK
  - Preferences > Search
  - Bookmarks manager
  - about:brave
  
The torrent viewer and torrent file list pages use the sortable tables too, but I'm not familiar with how to pull them up. @alexwykoff, @srirambv, or @luixxiul - can one of you help me with that?

File: less/about/preferences.less [hunk 1]
```diff
@@ -490,23 +490,30 @@ table.sortableTable {
     td {
       padding: 0 15px;
     }
+  }
 
-    td {
-      position: relative;
```
Comment (luixxiul) [inline less/about/preferences.less]: removing this breaks the position of the verified icon.

**Comment by bsclifton** [comment]
@jkup regarding the test- I tried it out... I think the test is written correct; it probably broke though when the default time for a visit was bumped up.  We can likely increase that timeout to fix the issue.  Since it's failing on master, I made an issue to track it:
https://github.com/brave/browser-laptop/issues/6110

**Comment by jkup** [comment]
@bsclifton is this one ready to merge?

**Comment by bsclifton** [comment]
@jkup yup! looks good to me :smile:

**Comment by luixxiul** [comment]
Was the position of verified icon confirmed?

**Comment by bsclifton** [comment]
@luixxiul thanks for the reminder! No- it still does not look correct

@jkup can you fix it up please? :smile: 
screen shot 2016-12-14 at 10 12 05 am

**Comment by bsclifton** [comment]
@jkup can you rebase? thanks :smile:

**Comment by jkup** [comment]
Done. I ran into a lot of merge conflicts but I checked all our sortable tables and they appear to look right!

File: less/about/preferences.less [hunk 2]
```diff
@@ -1170,6 +1196,33 @@ table.sortableTable {
     }
   }
 
+  .sort {
+    width: 100%;
+    text-align: left;
+  }
+
+  #paymentHistory {
```
Comment (luixxiul) [inline less/about/preferences.less]: ditto.

**Comment by jkup** [comment]
Thanks @luixxiul! Updated.

**Comment by luixxiul** [comment]
This change causes some regressions on about:preferences#payments. I will make it more specific, adding notes in preferences.less

**Comment by jkup** [comment]
Yeah I had put this PR up before you made all those design tweaks to prefs#payments. If you could let me know anything specific that would be great and I'll take care of it.

**Comment by luixxiul** [comment]
@jkup thanks, I'll add a commit soon. Please don't squash it and let @bsclifton review it ;-)

**Comment by luixxiul** [comment]
Pushing it, rebasing against master branch.

File: less/about/preferences.less [hunk 3]
```diff
@@ -1162,71 +1188,108 @@ table.sortableTable {
...
+      margin-top: @walletBarItemMargin;
     }
 
     .nextReconcileDate {
       margin-bottom: 0;
     }
   }
 
+  // This defines the styling inside the table (3rd) on about:preferences#payments
   .ledgerTable {
```
Comment (luixxiul) [inline less/about/preferences.less]: @jkup I moved your changes under here, `.ledgerTable`.

**Comment by luixxiul** [comment]
Still we need some polish, for example:

Actual:
screenshot 2016-12-17 16 27 11

Expected:
screenshot 2016-12-17 16 29 18

**Comment by luixxiul** [comment]
I'm going to fix it like this:

screenshot 2016-12-17 18 16 13

I'm wondering if these could be possible:

- Replace "Remove" with the icon (because the 6 characters can be longer in other languages and we do not want it)
- Add redundant `<tr>` to add margin between `<tbody>` and the first `<tr>`
  cf: http://stackoverflow.com/a/10690355

**Comment by luixxiul** [comment]
Also "Never include this site" in the context menu should be removed as it can be replaced with the icon.

screenshot 2016-12-17 18 57 27

**Comment by bradleyrichter** [comment]
"X" might work to replace "remove" if the icon or font awesome option will delay this PR.

**Comment by bsclifton** [comment]
I just rebased (no changes made) since critical fixes like the burger menu not closing were needed to properly test this.  Everything looks really good :smile: 

Here's a screenshot of how the verified check looks now:
screen shot 2016-12-19 at 2 26 02 pm

**Comment by bsclifton** [comment]
@luixxiul there is a CSS/LESS conflict here; can you please resolve this (either in GitHub's UI or by checking out / rebasing via command line).  Thanks :smile:

**Comment by luixxiul** [comment]
Rather this PR can be closed as I think this has been superseded with #6422, where those two commits were cherry-picked.

**Comment by luixxiul** [comment]
If so, let's merge the test plan described in the first post here with that of #6422.

**Comment by bsclifton** [comment]
@luixxiul let's keep 'em separate for now. When it's "go time", we'll want to merge this one first

**Comment by luixxiul** [comment]
Ah right, we can remove the cherry-picked commits and I've totally forgotten about that! :D

**Comment by bsclifton** [comment]
This one is failing tests:
`npm run test -- --grep="synopsis can disable site"`

**Comment by jkup** [comment]
@bsclifton test is fixed!

@luixxiul let's try not to add commits to existing PRs. Should your CSS commit be rebased into mine? If so, can you rebase the now 3 commits into one? If not, maybe pull yours out into a separate ticket so I can rebase my two?

**Comment by luixxiul** [comment]
@jkup you could drop my 886cf71 of this PR as it was cherry-picked https://github.com/brave/browser-laptop/pull/6422/commits/7ccae16bc3a3ffe41d792c4f28b27c1788b7e948 of #6422. / cc: @cezaraugusto

**Comment by bsclifton** [comment]
[Last travis-ci run](https://travis-ci.org/brave/browser-laptop/builds/190835561) shows 2 unrelated about:bookmark errors; I re-ran those locally and it's working as expected :smile: Merging...

**Pull request merged**